src/parseTable.ts

```diff
@@ -193,16 +201,36 @@ function getDateString(dateRaw: string | undefined): string | undefined {
  *
  * First, `/` and `\` are removed since they can't exist in file path.
  * Second, if the escaped url is a empty string or user doesn't specify an
- * url, use page id as the url.
- * @param {Nast.Page} page
- * @param {string} propId
+ * url, generate a slug from the title (if `autoSlug` is `true` in `Config`)
+ * and use it along with page id as the filename,
+ * otherwise just use the page id as is.
+ * @param {string} pageUri
+ * @param {string} customSlug
+ * @param {string} title
+ * @param {Config} config
+ * @returns {string}
+ */
+function getPageUrl(pageUri: string, customSlug: string, title: string, config: Config): string {
+  var url = getSafeUrl(customSlug)
+  if (url.length == 0) {
+    if (config.get("autoSlug")) {
+      const partialId = extractIdFromUri(pageUri).slice(0, 6);
+      url = `${getSlugFromTitle(title)}-${partialId}`
+    } else {
+      url = `${extractIdFromUri(pageUri)}`
+    }
+  }
+  return `${url}.html`
+}
+
+/**
+ * Returns a formatted slug from a title by stripping non-alphanumeric chars, and
+ * replacing spaces with dashes.
+ * @param {string} title 
  * @returns {string}
  */
-function getPageUrl(uri: string, slug: string): string {
-  const safeUrl = getSafeUrl(slug)
-  const realUrl =
-    safeUrl.length > 0 ? `${safeUrl}.html` : `${extractIdFromUri(uri)}.html`
-  return realUrl
+function getSlugFromTitle(title: string): string {
+  return title.replaceAll(/\s/g, "-").replaceAll(/[^\w-]/g, "").toLowerCase();
```

Comment:
A small opinion:

```js
"Learn Web Programming: HTML/CSS/JS 中文".replaceAll(/[^\w]/g, "-").toLowerCase()
// result: 'learn-web-programming--html-css-js---'
```
v.s.
```js
"Learn Web Programming: HTML/CSS/JS 中文".replaceAll(/\s/g, "-").replaceAll(/[^\w-]/g, "").toLowerCase()
// result: 'learn-web-programming-htmlcssjs-'
```

I prefer `HTML/CSS/JS` be converted to `html-css-js`, what do you think?

Reply:
For the specific case of `HTML/CSS/JS` becoming `html/css/js` I agree -- but the extra dashes everywhere else `-` look unsightly to me imo.

How about apart from whitespaces, we also convert slashes `/` and `\`? (And maybe we can choose more symbols to allowlist if we want to)

```js
"Learn Web Programming: HTML/CSS/JS 中文".replaceAll(/[\s\\\/]/g, "-").replaceAll(/[^\w-]/g, "").toLowerCase()
// result: 'learn-web-programming-html-css-js-'
```

How about that?

Reply:
It looks good to me!